durable-storage/src/merkle_layer/node.rs

```diff
@@ -0,0 +1,194 @@
+// SPDX-FileCopyrightText: 2025 Trilitech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+use std::fmt::Debug;
+
+use super::Key;
+
+/// A node which supports rebalancing operations
+pub(super) trait AvlNode: BinaryTreeNode + BinaryTreeNodeInvalidating {
+    /// Delete the node with a given key.
+    ///
+    /// If it does not exist, do nothing.
+    fn delete(_root: &mut Option<Box<Self>>, _key: &Key) {
+        todo!()
+    }
+
+    /// Return the height of the branch including this node.
+    #[cfg(test)] // Currently only used in a test
+    fn height(&self) -> u32;
+
+    /// Rebalance the node so that the difference in height between child branches is in the range
+    /// of -2..=2.
```

Comment:
```suggestion
    /// of -1..=1.
```

Or is this referring to something else?